src/main/java/com/code_intelligence/jazzer/driver/junit/JUnitRunner.java

```diff
@@ -0,0 +1,146 @@
+                                .addTestEngines(new JupiterTestEngine())
+                                .enableLauncherDiscoveryListenerAutoRegistration(false)
+                                .enableLauncherSessionListenerAutoRegistration(false)
+                                .enablePostDiscoveryFilterAutoRegistration(false)
+                                .enableTestEngineAutoRegistration(false)
+                                .enableTestExecutionListenerAutoRegistration(false)
+                                .build();
+
+    Map<String, String> indexedArgs =
+        IntStream.range(0, libFuzzerArgs.size())
+            .boxed()
+            .collect(Collectors.toMap(i -> "jazzer.internal.arg." + i, libFuzzerArgs::get));
+
+    LauncherDiscoveryRequestBuilder requestBuilder =
+        LauncherDiscoveryRequestBuilder.request()
+            .configurationParameter("jazzer.internal.commandLine", "true")
+            .configurationParameters(indexedArgs)
+            .selectors(selectClass(testClassName))
+            .filters(includeTags("jazzer"));
+    if (!Opt.targetMethod.isEmpty()) {
+      // HACK: This depends on JUnit internals as we need to filter by method name without having to
+      // specify the parameter types of the method.
+      requestBuilder.filters((PostDiscoveryFilter) testDescriptor
+          -> includedIf(!(testDescriptor instanceof MethodBasedTestDescriptor)
+              || ((MethodBasedTestDescriptor) testDescriptor)
+                     .getTestMethod()
+                     .getName()
+                     .equals(Opt.targetMethod)));
+    }
+    LauncherDiscoveryRequest request = requestBuilder.build();
+    Launcher launcher = LauncherFactory.create(config);
+    TestPlan testPlan = launcher.discover(request);
+    if (!testPlan.containsTests()) {
+      return Optional.empty();
+    }
+    return Optional.of(new JUnitRunner(launcher, testPlan));
+  }
+
+  public int run() {
+    AtomicReference<TestExecutionResult> resultHolder =
+        new AtomicReference<>(TestExecutionResult.successful());
+    launcher.execute(testPlan, new TestExecutionListener() {
+      @Override
+      public void executionFinished(
+          TestIdentifier testIdentifier, TestExecutionResult testExecutionResult) {
+        // Lifecycle methods can fail too, which results in failed execution results on container
+        // nodes. We keep the last failing one.
+        if (testExecutionResult.getThrowable().isPresent()) {
+          resultHolder.set(testExecutionResult);
+        }
+      }
+
+      @Override
+      public void reportingEntryPublished(TestIdentifier testIdentifier, ReportEntry entry) {
+        entry.getKeyValuePairs().values().forEach(Log::info);
+      }
+    });
+
+    TestExecutionResult result = resultHolder.get();
+    if (result.getStatus() != TestExecutionResult.Status.FAILED) {
+      return 0;
+    }
+
+    Throwable throwable = result.getThrowable().get();
```

Comment:
Do all failed tests have a `Throwable`? Otherwise `get` could throw an exception.

Reply:
They do, added a comment with a reference and restructured the logic to make this more clear.